layers/+lang/python/packages.el

```diff
@@ -105,6 +105,15 @@
         "hh" 'anaconda-mode-show-doc
         "gu" 'anaconda-mode-find-references))))
 
+(defun python/post-init-window-purpose ()
+  (add-hook 'python-mode-hook (lambda ()
+                                (spacemacs|window-purpose-define-config
+                                 python (:name-purposes '(("*compilation*" . compile))
```

Comment:
The compilation buffer isn't "owned" by the python layer, so this is a bad place to configure its purpose. Also, it's better to configure by mode than by name when possible, so in this case use `compilation-mode` instead of `"*compilation*"`.

Note that `compilation-mode` already has a `search` purpose (a bad name for a purpose, but that's an upstream issue), but we can override it.

Reply:
Ups, didn't thought about the compilation buffer's owner status. 
Ok I will use modes instead of names.